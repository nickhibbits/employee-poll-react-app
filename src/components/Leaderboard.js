import { connect } from "react-redux";
import "../styles/leaderboard.css";
import LeaderboardRow from "./LeaderboardRow";

const LeaderBoard = ({ users }) => {
  console.log("users", users);

  function getSum(a, b) {
    return a + b;
  }
  return (
    <div className="leaderboard-component">
      <h2>Leaderboard</h2>
      <div className="table-wrapper">
        <table className="leaderboard-table">
          <tr className="leaderboard-title-wrapper">
            <th className="user-title">User</th>
            <th className="questions-asked-title">Asked</th>
            <th className="questions-answered-title">Answered</th>
          </tr>
          {Object.values(users)
            .sort((a, b) => {
              return (
                getSum(b.questions.length, Object.keys(b.answers).length) -
                getSum(a.questions.length, Object.keys(a.answers).length)
              );
            })
            .map((user) => {
              return <LeaderboardRow key={user.id} user={user} />;
            })}
        </table>
      </div>
    </div>
  );
};

const mapStateToProps = ({ users }) => ({
  users,
});
export default connect(mapStateToProps)(LeaderBoard);
