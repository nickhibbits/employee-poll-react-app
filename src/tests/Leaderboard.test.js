import Leaderboard from "../components/leaderboard/Leaderboard";
import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../store";

describe("Leaderboard", () => {
  it("will match snapshot", () => {
    var component = render(
      <Provider store={store}>
        <Leaderboard />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
