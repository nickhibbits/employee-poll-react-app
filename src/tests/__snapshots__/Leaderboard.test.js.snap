// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="leaderboard-component"
      >
        <h2>
          Leaderboard
        </h2>
        <div
          class="table-wrapper"
        >
          <table
            class="leaderboard-table"
          >
            <tbody>
              <tr
                class="leaderboard-title-wrapper"
              >
                <th
                  class="user-title"
                >
                  User
                </th>
                <th
                  class="questions-asked-title"
                >
                  Asked
                </th>
                <th
                  class="questions-answered-title"
                >
                  Answered
                </th>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="leaderboard-component"
    >
      <h2>
        Leaderboard
      </h2>
      <div
        class="table-wrapper"
      >
        <table
          class="leaderboard-table"
        >
          <tbody>
            <tr
              class="leaderboard-title-wrapper"
            >
              <th
                class="user-title"
              >
                User
              </th>
              <th
                class="questions-asked-title"
              >
                Asked
              </th>
              <th
                class="questions-answered-title"
              >
                Answered
              </th>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
